### DISCUSSION TITLE
Using Google calendar for automatic switching

### DISCUSSION THREAD

**Discussion by on-off-on** [arendst/Tasmota#12324] [Ideas]
Hi, I think there is a marvelous function for Tasmoto:

"automatic switching using Google calendar"

This feature is also available for AVM!200 and the feature is really great. 

Add an entry on google calander and the device acts as requested.

For details pls search/check "Using Google calendar for automatic switching" and "AVM"

What do you think about the feature for upcomming tasmoto version?

**Comment by Jason2866** [comment]
Thia is not the goal of Tasmota. It is designed to be controlled via mqtt.
The hardware is a very limited device, every implemented Feature needs resources....
Every good HA system supports mqtt. So this is the place to do "magic things"
and send the commands to the devices.

**Comment by on-off-on** [comment]
Just a question concerning ESP8266 and "automatic switching using Google calendar"

Has anybody information about such a solution for ESP8266? 

I´m one of the happy user which use the on the AVM!200 implemented feature but if possible I would like to replace all my AVM200 switches with an ESP8266 or best case tasmota solution.

So... if anybody has a idea -please leave a message.

**Comment by jwosullivan** [comment]
Have you seen this video from Andreas Spiess?

https://www.youtube.com/watch?v=ZFgh2KFLeGA

it doesn’t use tasmota

Regards,

James

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by sourcesimian** [comment]
@on-off-on I also wanted to switch things on and off as scheduled in Google Calendars with an ESP8266 but could not find a library so I wrote one: https://github.com/sourcesimian/uICAL, perhaps you might be interested in this: https://github.com/sourcesimian/uICAL/blob/master/examples/sonoff-basic/README.md

**Comment by ascillato2** [comment]
@sourcesimian

Great!!

Thanks for sharing!!!
